launch/launch/substitutions/launch_configuration.py

```diff
@@ -41,8 +40,6 @@ def __init__(
         """Constructor."""
         super().__init__()
 
-        ensure_argument_type(variable_name, str, 'variable_name', 'LaunchConfiguration')
```

Comment:
@ivanpauno shouldn't we ensure it's `SomeSubstitutionsType` anyways?

Reply:
We aren't usually ensuring that arguments are of an specific type. e.g.:
https://github.com/ros2/launch/blob/d2ea92cc1136eb655f3a61f3a01cd2ae2740a5aa/launch/launch/substitutions/environment_variable.py#L26-L44
I don't know why we should do it here and not in other places.

CI is already green or with unrelated errors. Could you please check macOS failures too? They seem to be unrelated.

Reply:
Alright, fair enough.